### PR TITLE
Fix #104: Protect API behind authentication

### LINKED ISSUE #104: Protect API endpoints behind authentication
Require authentication to load /metrics/, /metric/#/, etc.

### PR DESCRIPTION
**Pull request by openjck**

### CODE REVIEW

File: distributionviewer/authentication.py [hunk 1]
```diff
@@ -1,75 +0,0 @@
...
-    unique_id = os.urandom(8)
-
-    consumer_id = hashlib.sha1(
-        email + settings.SECRET_KEY).hexdigest()
-
-    hm = hmac.new(
-        unique_id + settings.SECRET_KEY,
-        consumer_id, hashlib.sha512)
-    return u','.join((email, hm.hexdigest().decode('ascii'),
-                      unique_id.encode('base64').decode('ascii')))
```
Comment (robhudson) [inline distributionviewer/authentication.py:75]: Thanks for removing. 👍

File: distributionviewer/api/views.py [hunk 2]
```diff
@@ -32,8 +32,8 @@ def render_numeric(dist):
     return s.data
 
 
+@login_required
```
Comment (robhudson) [inline distributionviewer/api/views.py]: Since you've already updated the default DRF settings to include `IsAuthenticated`, is this needed?

Comment (openjck) [reply distributionviewer/api/views.py]: That's a good question. As it turns out, the decorator isn't required but it does change the behavior. When the decorator isn't present, the API outputs the following:

```
{"metrics":{"detail":"Authentication credentials were not provided."}}
```

When the decorator is present, the user is redirected to the login page when trying to access API endpoints.

But you probably already knew that. I'm happy either way. Which do you think would be better?

File: distributionviewer/authentication.py (hunk 1 above)
Comment (openjck) [reply distributionviewer/authentication.py:75]: Thank @washort! He's the one who realized we didn't need  it. :smile:

File: distributionviewer/api/views.py (hunk 2 above)
Comment (robhudson) [reply distributionviewer/api/views.py]: I think for an API a 403 with detail is better. I wouldn't expect a redirect from an API response.

Comment (openjck) [reply distributionviewer/api/views.py]: Okay, great! I'll make that update.

**Comment by openjck** [comment]
Updated

**Pull request merged**